pluginops/prs.go

```diff
@@ -0,0 +1,249 @@
+				out += Green("\u2713")
+			} else {
+				out += Red("\u2715")
+			}
+			fmt.Print(out + "\n")
+		}
+
+		return nil
+	},
+}
+
+var prsMergeCmd = &cobra.Command{
+	Use:   "merge",
+	Short: "Merge plugin PRs that are ready.",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		client, err := getGitHubClient()
+		if err != nil {
+			return err
+		}
+
+		ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+		defer cancel()
+
+		prs, err := getPluginPRs(ctx, client)
+		if err != nil {
+			return err
+		}
+
+		if len(prs) == 0 {
+			fmt.Println("No PRs to merge.")
+			return nil
+		}
+
+		slices.SortFunc(prs, func(a, b *prInfo) bool { return a.Repository < b.Repository })
+
+		for _, pr := range prs {
+			if !pr.readyToBeMerged() {
+				log.WithFields(log.Fields{"repo": pr.Repository, "title": pr.GetTitle()}).Debug("PR not ready to be merged")
+				continue
+			}
+
+			ok, err := confirmPrompt(fmt.Sprintf("Merge %s/%s", pr.Repository, pr.GetTitle()))
+			if err != nil {
+				return err
+			}
+			if !ok {
+				continue
+			}
+
+			err = mergePR(context.Background(), client, pr)
+			if err != nil {
+				return err
+			}
+
+			fmt.Printf("Merged %s\n\n", pr.GetHTMLURL())
+		}
+
+		return nil
+	},
+}
+
+type prInfo struct {
+	Repository string
+	*github.PullRequest
+	reviews []*github.PullRequestReview
+}
+
+func getPluginPRs(ctx context.Context, client *github.Client) ([]*prInfo, error) {
+	user, _, err := client.Users.Get(ctx, "")
+	if err != nil {
+		return nil, err
+	}
+
+	query := fmt.Sprintf("is:pr is:open author:%v org:%v", user.GetLogin(), org)
+
+	issues, _, err := client.Search.Issues(ctx, query, &github.SearchOptions{}) // TODO: handle pagination
+	if err != nil {
+		return nil, err
+	}
+
+	var prs []*prInfo
+
+	var wg sync.WaitGroup
+	for _, issue := range issues.Issues {
+		if issue.IsPullRequest() {
+			url := issue.GetHTMLURL()
+			url = strings.TrimPrefix(url, "https://github.com/")
+			url = strings.TrimPrefix(url, org+"/")
+
+			repo := strings.Split(url, "/")[0]
+
+			if !isPluginRepo(repo) {
+				continue
+			}
+
+			wg.Add(1)
+			go func(repo string, number int) {
+				defer wg.Done()
+				pr, _, err := client.PullRequests.Get(ctx, org, repo, number)
+				if err != nil {
+					log.WithError(err).Warn("failed to get PR")
+				}
+
+				reviews, _, err := client.PullRequests.ListReviews(ctx, org, repo, number, &github.ListOptions{}) // TODO: handle pagination
+				if err != nil {
+					log.WithError(err).Warn("failed to get PR")
+				}
+				prs = append(prs, &prInfo{
+					Repository:  repo,
+					PullRequest: pr,
+					reviews:     reviews,
+				})
+
+			}(repo, issue.GetNumber())
+
+		}
+	}
+
+	wg.Wait()
+
+	return prs, nil
+}
+
+func (pr *prInfo) readyToBeMerged() bool {
+	for _, l := range pr.Labels {
+		if l.GetName() == "Do Not Merge" {
+			return false
+		}
+		if l.GetName() == "Work In Progress" {
+			return false
+		}
+		if l.GetName() == "3: QA Review" {
+			return false
+		}
+	}
```

Comment:
cc @DHaussermann for visibility. This function is used to determine if a given PR is a candidate to be merged through this tool's automation